pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/startree/AggregationSpec.java

```diff
@@ -82,6 +82,15 @@ public Map<String, Object> getFunctionParameters() {
     return _functionParameters;
   }
 
+  /// Returns `true` if the given [AggregationSpec] does not match the current one, in which case the star-tree index
+  /// needs to be updated, {@code false} otherwise.
+  /// - Update star-tree if the function parameters or compression codec changes.
+  /// - Do not update star-tree when index version or config changes to avoid rebuilding all the star-trees when the
+  /// default index setting changes. This is consistent with the behavior of forward indexes.
+  public boolean shouldModifyStarTree(AggregationSpec that) {
```

Comment:
what about configs like targetDocsPerChunk, targetMaxChunkSize and deriveNumDocsPerChunk? 

btw why compressionCodec has to be checked?

Reply:
This is essentially making the behavior consistent with forward index. See `ForwardIndexHandler.shouldChangeRawCompressionType()` for more context.
We can revisit this and potentially adding a flag to control the behavior.

Reply:
I see. Actually, I just realized that `config changes` as you mentioned in the comment must be these three configs: "targetDocsPerChunk, targetMaxChunkSize and deriveNumDocsPerChunk".

It may help to add in the comment how to force the version/config changes if wanted.